taichi/transforms/demote_dense_struct_fors.cpp

```diff
@@ -17,27 +16,26 @@ VecStatement convert_to_range_for(StructForStmt *struct_for) {
     snode = snode->parent;
   }
   std::reverse(snodes.begin(), snodes.end());
-  TI_ASSERT(total_bits <= 31);
+  TI_ASSERT(total_bits <= 30);
 
   auto upper_bound = 1 << total_bits;
```

Comment:
`1 << 31` looks dangerous. So I suppose not supporting tensors of size 2^31 for now.